src/test/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,175 @@
+import org.apache.mahout.cf.taste.model.DataModel;
+import org.apache.mahout.cf.taste.neighborhood.UserNeighborhood;
+import org.apache.mahout.cf.taste.recommender.RecommendedItem;
+import org.apache.mahout.cf.taste.recommender.UserBasedRecommender;
+import org.apache.mahout.cf.taste.similarity.UserSimilarity;
+
+import java.io.FileWriter;
+import java.io.File;
+import java.io.InputStream;
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.Reader;
+import java.io.InputStreamReader;
+import java.io.IOException;
+
+import java.util.zip.GZIPInputStream;
+import java.util.List;
+import java.util.Map;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+class MovieRecommender {
+    private ArrayList<Double> reviews = new ArrayList<Double>();
+    private ArrayList<String> users = new ArrayList<String>();
+    private ArrayList<String> products = new ArrayList<String>();
+
+    private Map<String,Integer> usersMap = new HashMap<String, Integer>();
+    private Map<String, Integer> productsMap = new HashMap<String, Integer>();
+
+
+    MovieRecommender(String path){
+        this.readTxtFile(path);
+    }
+
+
+    private void readTxtFile(String path){
+        String userKey = "review/userId: ";
+        String productKey = "product/productId: ";
+        String reviewKey = "review/score: ";
+
+        InputStream fileStream;
+
+        try{
+            fileStream = new FileInputStream(path);
+            InputStream gzipStream = new GZIPInputStream(fileStream);
+            Reader decoder = new InputStreamReader(gzipStream, "UTF-8");
+            BufferedReader buffered = new BufferedReader(decoder);
+
+            int usersCounter = 1;
+            int productsCounter = 1;
+            String content;
+
+            while ((content = buffered.readLine()) != null){
+                if (content.contains(userKey)){
+                    String userId = content.split(userKey)[1];
+                    users.add(userId);
+                    if(!this.usersMap.containsKey(userId)){
+                        this.usersMap.put(userId, usersCounter);
+                        usersCounter++;
+                    }
+                }
+                else if(content.contains(productKey)){
+                    String productId = content.split(productKey)[1];
+                    products.add(productId);
+                    if(!this.productsMap.containsKey(productId)){
+                        this.productsMap.put(productId, productsCounter);
+                        productsCounter++;
+                    }
+                }
+                else if(content.contains(reviewKey)){
+                    String scoreValue = content.split(reviewKey)[1];
+                    reviews.add(Double.valueOf(scoreValue));
+                }
+            }
+
+            System.out.println("users: " + usersMap.size() + " products: " + productsMap.size() + " reviews: " + reviews.size());
```

Comment:
The String concatenation using this way is not too much recommendable, you could construct a String using StringBuilder; in this case is just one String, but you may consider that :) Also, you can implement a Logger here instead of the sout